frontend/src/hooks/queryBuilder/useOptions.ts

```diff
@@ -52,41 +52,39 @@ export const useOptions = (
 	);
 
 	useEffect(() => {
+		let newOptions: Option[] = [];
+
 		if (!key) {
-			setOptions(
-				searchValue
-					? [
-							{ label: `${searchValue} `, value: `${searchValue} ` },
-							...getOptionsFromKeys(keys),
-					  ]
-					: getOptionsFromKeys(keys),
-			);
+			newOptions = searchValue
+				? [
+						{ label: `${searchValue} `, value: `${searchValue} ` },
+						...getOptionsFromKeys(keys),
+				  ]
+				: getOptionsFromKeys(keys);
 		} else if (key && !operator) {
-			setOptions(
-				operators?.map((operator) => ({
-					value: `${key} ${operator} `,
-					label: `${key} ${operator} `,
-				})),
-			);
+			newOptions = operators?.map((operator) => ({
+				value: `${key} ${operator} `,
+				label: `${key} ${operator} `,
+			}));
 		} else if (key && operator) {
 			if (isMulti) {
-				setOptions(
-					results.map((item) => ({
-						label: checkCommaInValue(String(item)),
-						value: String(item),
-					})),
-				);
+				newOptions = results.map((item) => ({
+					label: checkCommaInValue(String(item)),
```

Comment:
What is this `checkCommaInValue`? We shouldn't use naive string manipulations anywhere in the code. I have already suggested a fix here https://github.com/SigNoz/signoz/pull/2629#discussion_r1190850173

Reply:
This still needs to be fixed